src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -400,234 +443,223 @@ End Class
+                    false, 93)]
+        [InlineData("new byte[s.Length], 0, (int)s.Length, new CancellationToken()",
+                    "(new byte[s.Length]).AsMemory(0, (int)s.Length), new CancellationToken()",
+                    true, 93)]
+        public Task CS_Analyzer_Diagnostic_InlineByteArray(string originalArgs, string fixedArgs, bool withConfigureAwait, int endColumn)
+        {
+            string source = @"
 using System;
 using System.IO;
 using System.Threading;
 class C
-{
+{{
     public async void M()
-    {
-        using (FileStream fs = File.Open(""file.txt"", FileMode.Open))
-        {
-            byte[] buffer = new byte[fs.Length];
-            await fs.ReadAsync(buffer, 0, (int)fs.Length, new CancellationToken()).ConfigureAwait(false);
-        }
-    }
-}
-            ", GetCSResult(12, 19, 12, 105));
+    {{
+        using (FileStream s = File.Open(""file.txt"", FileMode.Open))
+        {{
+            await {0};
+        }}
+    }}
+}}
+            ";
+            return GetCSReadAsyncDiagnostic(source, originalArgs, fixedArgs, withConfigureAwait, 11, 19, 11, endColumn);
         }
 
         #endregion
 
         #region VB - Diagnostic - Analyzer
 
-        [Fact]
-        public Task VB_Analyzer_Diagnostic_ByteArray()
-        {
-            return AnalyzeVBAsync(@"
+        [Theory]
+        [InlineData("buffer, 0, s.Length",
+                    "buffer.AsMemory(0, s.Length)",
+                    false, 51)]
+        [InlineData("buffer, 0, s.Length",
+                    "buffer.AsMemory(0, s.Length)",
+                    true, 51)]
+        [InlineData("buffer, 0, s.Length, New CancellationToken()",
+                    "buffer.AsMemory(0, s.Length), New CancellationToken()",
+                    false, 76)]
+        [InlineData("buffer, 0, s.Length, New CancellationToken()",
+                    "buffer.AsMemory(0, s.Length), New CancellationToken()",
+                    true, 76)]
+        public Task VB_Analyzer_Diagnostic_VarByteArray(string originalArgs, string fixedArgs, bool withConfigureAwait, int endColumn)
+        {
+            string source = @"
 Imports System
 Imports System.IO
 Imports System.Threading
-Class C
+Public Module C
     Public Async Sub M()
-        Using fs As FileStream = File.Open(""file.txt"", FileMode.Open)
-            Dim buffer As Byte() = New Byte(fs.Length - 1) {}
-            Await fs.ReadAsync(buffer, 0, fs.Length)
+        Using s As FileStream = File.Open(""file.txt"", FileMode.Open)
+            Dim buffer As Byte() = New Byte(s.Length - 1) {{}}
+            Await {0}
         End Using
     End Sub
-End Class
-            ", GetVBResult(9, 19, 9, 53));
-        }
-
-        [Fact]
-        public Task VB_Analyzer_Diagnostic_Inline()
-        {
-            return AnalyzeVBAsync(@"
+End Module
+            ";
+            return GetVBReadAsyncDiagnostic(source, originalArgs, fixedArgs, withConfigureAwait, 9, 19, 9, endColumn);
```

Comment:
Ideally, we can just get rid of the last 4 arguments here by switching to markup syntax for code fix tests.

Reply:
See my answer above.